internal/storage/storage.go

```diff
@@ -0,0 +1,213 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"os"
+	"time"
+
+	// sqlite3 driver
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	querySaveSnapshot = `INSERT INTO snapshots(cluster_name, created_at, data) 
+							VALUES(?, ?, ?)`
+	querySaveRecoveries = `INSERT INTO recoveries(cluster_name, created_at, data) 
+							VALUES(?, ?, ?)`
+	queryCreateTableSnapshots = `CREATE TABLE IF NOT EXISTS snapshots (
+		"id" integer NOT NULL PRIMARY KEY AUTOINCREMENT,		
+		"cluster_name" TEXT,
+		"created_at" INTEGER,
+		"data" BLOB
+	  )`
+	queryCreateTableRecoveries = `CREATE TABLE IF NOT EXISTS recoveries (
+		"id" integer NOT NULL PRIMARY KEY AUTOINCREMENT,		
+		"cluster_name" TEXT,
+		"created_at" INTEGER,
+		"data" BLOB
+	  )`
+	queryGetLastSnapshot = `SELECT data
+		FROM snapshots
+		WHERE cluster_name = ?
+		ORDER BY id DESC limit 1`
+	queryGetRecoveries = `SELECT data
+		FROM recoveries
+		WHERE cluster_name = ?`
+)
+
+var (
+	ErrEmptyResult = errors.New("empty result")
+)
+
+type SaveRequest struct {
+	ClusterName string
+	CreatedAt   int64
+	Data        []byte
+}
+
+type Storage interface {
+	SaveSnapshot(context.Context, SaveRequest) error
+	SaveRecovery(context.Context, SaveRequest) error
+	GetClusterLastSnapshot(context.Context, string) ([]byte, error)
```

Comment:
It is quite strange to return raw bytes instead of DTO

Reply:
Replaced with DTO